main/src/ca/uwaterloo/flix/language/phase/unification/HybridFormula.scala

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2022 Magnus Madsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.phase.unification
+
+import ca.uwaterloo.flix.language.ast.{Symbol, Type}
+import ca.uwaterloo.flix.language.phase.unification.BddFormula.BddFormula
+import ca.uwaterloo.flix.util.InternalCompilerException
+import ca.uwaterloo.flix.util.collection.Bimap
+import org.sosy_lab.pjbdd.api.DD
+
+import scala.collection.immutable.SortedSet
+
+trait Formula {}
```

Comment:
Maybe HybridFormula?

Reply:
This is supposed to be a super-type for the types that are inside a HybridFormula so the formula inside can be either a BoolFormula or BddFormula. Since there is a class called HybridFormula, the trait cannot also be called that